python/ray/_private/runtime_env/plugin.py

```diff
@@ -247,12 +259,15 @@ async def create_for_plugin_if_needed(
         await plugin.create(None, runtime_env, context, logger=logger)
 
     for uri in uris:
-        if uri not in uri_cache:
-            logger.debug(f"Cache miss for URI {uri}.")
-            size_bytes = await plugin.create(uri, runtime_env, context, logger=logger)
-            uri_cache.add(uri, size_bytes, logger=logger)
-        else:
-            logger.debug(f"Cache hit for URI {uri}.")
-            uri_cache.mark_used(uri, logger=logger)
+        async with uri_locks[uri]:
```

Comment:
How to do gc for the uri locks?

Reply:
we use `WeakValueDictionary` to avoid handle the gc of uri_lock

Reply:
Nice, I didn't know about this